plugins/hls-tactics-plugin/README.md

```diff
@@ -34,6 +34,19 @@ fill hole" code action, *et voila!*
 [hls]: https://github.com/haskell/haskell-language-server/releases
 
 
+## Usage
+
+When enabled, Wingman for Haskell will remove HLS support for hole-fit code
+actions. These code actions are provided by GHC and make typechecking extremely
+slow in the presence of typed holes. Because Wingman relies so heavily on typed
+holes, these features are in great tension.
+
+The solution: we just remove the hole-fit actions. They are mostly subsumed by
+Wingman, and usually make extremely unhelpful suggestions, such as `replace _
+with id _`. If you'd prefer to use these actions, you can get them back by
+disabling Wingman.
```

Comment:
The "usually make extremely unhelpful suggestions" is mostly caused by the refinement hole fits (and the lack of filtering out `id`, `const`, `$`, from the list, but that wouldn't even be necessary most of the time without refinement hole fits), so that piece documentation would be pretty misleading. If we just remove the refinement holes, the feature becomes 1000% better.

Reply:
Yeah, but unfortunately removing just refinement holes doesn't solve the underlying issue. I suggested a feature in https://github.com/haskell/haskell-language-server/issues/1602#issuecomment-857819575 that would give us most of the best of both worlds.

Reply:
Losing the ability to discover things from external modules would severely cripple the feature, so I wouldn't quite see it as the best of both worlds, but at least it's better than nothing. :/